### DISCUSSION TITLE
[Bug]: attachment system behave abnormally

### DISCUSSION THREAD

**Discussion by Astricaelus** [danny-avila/LibreChat#2822] [Troubleshooting] [answered]
### What happened?

I'm using docker installation and when a few hours ago (2024-05-22 9:00 UTC+8) I run `git pull && docker compose pull && docker compose up -d` to upgrade to newest version, the attachment system seems to behave abnormally.

Files uploaded during previous versions cannot be used(UI report `Something went wrong. Here's the specific error message we encountered: Error connecting to server`)

When I delete files in "My Files" tag, the file will disappear in the UI. But when the web page is reloaded, they will show up again.  If the files are uploaded during latest version, it can be used in the chat, otherwise UI still reports `Something went wrong. Here's the specific error message we encountered: Error connecting to server`.

Even though, I can still upload new files and use them.

### Steps to Reproduce

1. use `git clone xxx && docker compose up` method to install LibreChat and upgrade directly from previous versions. (I found that pdf files uploaded during previous version cannot be used,`Something went wrong. Here's the specific error message we encountered: Error connecting to server`) 
2. Delete some files, close the windows and reopen the window. The deleted files will show up again
3. Try to use these deleted files in a chat, files uploaded during latest version can be used. But for files uploaded during previous versions, the UI reports `Something went wrong. Here's the specific error message we encountered: Error connecting to server`

### What browsers are you seeing the problem on?

Firefox, Chrome

### Relevant log output

```shell
When I tried to delete files, the error log is:
< {"level":"error","message":"Error deleting file Request failed with status code 500","stack":"Error: Request failed with status code 500\n    at deleteVectors (/app/api/server/services/Files/VectorDB/crud.js:36:11)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async Promise.allSettled (index 0)\n    at async processDeleteRequest (/app/api/server/services/Files/process.js:168:3)\n    at async /app/api/server/routes/files/files.js:60:5"}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Looking into it, might be from a recent change in this library: https://github.com/danny-avila/rag_api

**Reply by danny-avila** [reply, marked as answer]
Will be fixing this here: https://github.com/danny-avila/rag_api/pull/42

**Reply by Astricaelus** [reply]
OK, thanks a lot!

**Reply by Astricaelus** [reply]
I tried to use older version images but the bug still exists. Has current version done some permanent changes to the database? Or do I need to manually do something to fix the database?

**Reply by danny-avila** [reply]
> I tried to use older version images but the bug still exists. Has current version done some permanent changes to the database? Or do I need to manually do something to fix the database?

I merged a fix a few hours after your message, it should be good now

**Reply by Astricaelus** [reply]
OK, thanks
